### ISSUE TITLE
False positive for `Layout/SpaceBeforeBlockBraces` cop

### ISSUE THREAD

**Issue by koic** [bbatsov/rubocop#5221]
This report is using RuboCop master (a28c9b2) .
https://github.com/bbatsov/rubocop/commit/a28c9b22f7d59f7b8a47fe71cb83d44c9c527743

I think that it is probably a false positive, but it is confirmed just in case.
AFAIK, it seems to have been changed with the following commit.
https://github.com/bbatsov/rubocop/pull/4926/files#diff-e93280b3b31a6438c533a5f3232340d8

The following is an excerpt of the difference of the config/default.yml file.

```diff
 Layout/SpaceBeforeBlockBraces:
   EnforcedStyle: space
    SupportedStyles:
      - space
      - no_space
 +  EnforcedStyleForEmptyBraces: no_space
    SupportedStylesForEmptyBraces:
      - space
      - no_space
```

## Expected behavior

It is the behavior of RuboCop 0.51.0.

No offenses.

## Actual behavior

It is the behavior of RuboCop 0.60.0? beta (master branch a28c9b2) .

The offense described in the reproduction is detected.

## Steps to reproduce the problem

Put a space between `foo` and `{}`.

```console
% cat /tmp/foo.rb
# frozen_string_literal: true

foo {}
```

Run RuboCop built with master branch (a28c9b2) .

```console
% rubocop /tmp/foo.rb
Inspecting 1 file
C

Offenses:

/tmp/foo.rb:3:4: C: Layout/SpaceBeforeBlockBraces: Space detected to the left of {.
foo {}
   ^

1 file inspected, 1 offense detected
```

This seems not to be the intended behavior for changes that added `EnforcedStyleForEmptyBraces: no_space` to config/default.yml.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
% rubocop -V
0.51.0 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-darwin13)
```

__This is not the released RuboCop 0.51.0. It is the master branch (a28c9b2) under development.__

**Comment by koic** [comment]
This false positive is detected in RuboCop 0.52.0.

**Comment by Drenmi** [comment]
Not really a false positive, just the wrong default value. Can't remember why we set it to `no_space`, but it had something to do with lambdas. 🤔 But I definitely think the default should be `space`.

**Comment by koic** [comment]
Oh, I got it. Thanks for the answer.

**Comment by mikecmpbll** [comment]
it would make more sense if the default was whatever the EnforcedStyle is. it's confusing to debug when you've set the Layout/SpaceBeforeBlockBraces EnforcedStyle to no_space yet rubocop flags up there being no space before block braces as an issue.
